analyzers/src/SonarAnalyzer.Common/Extensions/AttributeDataExtensions.cs

```diff
@@ -30,5 +32,47 @@ public static class AttributeDataExtensions
 
         public static bool HasAnyName(this AttributeData attribute, params string[] names) =>
             names.Any(x => attribute.HasName(x));
+
+        public static bool TryGetAttributeValue<T>(this AttributeData attribute, string valueName, out T value)
+        {
+            // named arguments take precedence over constructor arguments of the same name. For [Attr(valueName: false, valueName = true)] "true" is returned.
+            if (attribute.NamedArguments.IndexOf(x => x.Key.Equals(valueName, StringComparison.OrdinalIgnoreCase)) is var namedAgumentIndex and >= 0)
+            {
+                return TryConvertConstant(attribute.NamedArguments[namedAgumentIndex].Value, out value);
+            }
+            if (attribute.AttributeConstructor.Parameters.IndexOf(x => x.Name.Equals(valueName, StringComparison.OrdinalIgnoreCase)) is var constructorParameterIndex and >= 0)
+            {
+                return TryConvertConstant(attribute.ConstructorArguments[constructorParameterIndex], out value);
+            }
+            value = default;
+            return false;
+
+            static bool TryConvertConstant(TypedConstant constant, out T value)
+            {
+                value = default;
+                if (constant.IsNull)
+                {
+                    return true;
+                }
+                if (constant.Value is T result)
+                {
+                    value = result;
+                    return true;
+                }
+                if (constant.Value is IConvertible)
+                {
+                    try
+                    {
+                        value = (T)Convert.ChangeType(constant.Value, typeof(T));
+                        return true;
+                    }
+                    catch (Exception ex) when (ex is FormatException or OverflowException)
```

Comment:
What about `InvalidCastException`? After that, we can go just
```suggestion
                    catch
```